puphpet/config.yaml.template

```diff
@@ -60,10 +71,20 @@ vagrantfile-local:
         host: detect
 server:
     install: '1'
+    apt-ppa-repos:
+        - ppa:git-core/ppa   # Provides newer git-core. -bp
+        - ppa:chris-lea/node.js  # Provides newer node and npm. -bp
     packages:
-        - vim
-        - memcached
         - curl
+        - emacs
```

Comment:
I don't think rebuilding boxes is the end of the world. It's especially not a "huge" inconvenience if you're only building the box slowly once. If anything, that first slow build will be a plenty good reminder to trim up your config like you should be doing anyway. 

And on that point; we are professional, responsible developers, right? Those configs should always be an accurate representation of what the project needs, so naturally the first thing you'd be doing on a fresh project is beginning to customize it. Isn't that exactly what we did before we had the token replacement, after all? We changed the readme title from "title goes here" to "acme magic app". We haven't gotten rid of that step, we've just streamlined it. Maybe you want to prompt the user for each list of packages to include or exclude?? We could probably write more of puphpet.com into our skeleton too. :-p

Again, the point is to act as a primary reference for the company. We are trying to develop an unofficial standard that we all start from and largely conform to. It makes a ton of sense to me for the skeleton to lay out which packages are "suggested". 

Take having memcache AND redis in there like I do now: I'd be all for reducing down to one and forcing devs to intentionally have to switch to the other, thereby subtly encouraging the use of one over the other whenever possible. Likewise, I want the skeleton to _at least_ suggest I use memcache or redis over ALL other alternatives. 

You might not see the value of this now, but we are building towards having production infrastructures mirror or reflect these configs. So 

1. Just because you don't use it locally NOW doesn't mean it shouldn't be in the box (since if you are using it in production you SHOULD be using locally too) and 

2. Standardizing on a reasonable "maximum" list of stock dependencies means we can build ready to go **static** production images that will fit _all_ of those use cases without further customization (and work!)

We are trying to reduce our sysadmin overhead, not increase it.

Sent from my iPhone

> On Jan 31, 2015, at 10:52 AM, Andrew Lechowicz <notifications@github.com> wrote:
> 
> In puphpet/config.yaml.template:
> 
> >          - curl
> > +        - emacs
> If the process for a new project is composer create-project and vagrant up I this is is a huge inconvenience. Even if you knew you wanted to delete lines the process becomes composer create-project, scan though lines of code deleting things you may not need afraid to remove too much in case you break the puppet build, then vagrant up and hope you didn't remove something like intl and have to add it back and re-provision the box. This isn't event worst case scenario. I could see someone making an initial commit with all of this in the repo only to have to make another commit down the line to remove it all and re-buld the box and have every other dev do the same only to add one or two things back when the client asks for a feature that requires gd or geoip. composer create-project and vagrant up seems much more sane as I'm much more likely to know "I need imagick" and think "I better check that it will be installed" versus "I don't need, g++, graphviz, make..." and think "I better make sure those are not going to be installed".
> 
> —
> Reply to this email directly or view it on GitHub.
>